google/cloud/internal/log_impl.cc

```diff
@@ -26,17 +27,24 @@ void StdClogBackend::Process(LogRecord const& lr) {
   if (lr.severity >= Severity::GCP_LS_WARNING) {
     std::clog << std::flush;
   }
+  if (lr.severity == Severity::GCP_LS_FATAL) {
```

Comment:
Consider an alternative implementation.  Create a new `LogBackend` which *only* calls terminate if the severity is `GCP_LS_FATAL`, and initialize it here:

https://github.com/googleapis/google-cloud-cpp/blob/8f6afc3c9d503ebb634f189fd94f915c2e308b34/google/cloud/log.cc#L200

Maybe we can split `GOOGLE_CLOUD_CPP_EXPERIMENTAL_LOG_CONFIG` by `:` first, so people can write:

```
export GOOGLE_CLOUD_CPP_EXPERIMENTAL_LOG_CONFIG=terminate-on-fatal:clog
```

or 

```
export GOOGLE_CLOUD_CPP_EXPERIMENTAL_LOG_CONFIG=terminate-on-fatal:lastN,20,ERROR
```

What do you think?

Reply:
By "`GOOGLE_CLOUD_CPP_EXPERIMENTAL_LOG_CONFIG=terminate-on-fatal`", are you suggesting the behavior should be conditional?

Reply:
Hmmm... yes, I was thinking that.  I see where that would be weird or wrong.  Maybe we can start with the behavior being opt-in (after all, it does not crash today!) and then move to opt-out?

Reply:
Given that all but two of the existing `GCP_LOG(FATAL)` calls are immediately followed by `std::exit()`, and the intuition that in those other two cases the author thought it would not return, I'm not sure that we need such a migration strategy.

If my intuition is wrong (we can check with the authors), then perhaps an easier thing to do would be to just change those two to `GCP_LOG(ERROR)` first.

Reply:
I should have looked, thanks for doing so.  Yes, unconditional without migration SGTM then.